From 75095746b0b7e2b448d0a8847c9ec69fb7b8364e Mon Sep 17 00:00:00 2001
From: Andrea della Porta <andrea.porta@suse.com>
Date: Tue, 16 Sep 2025 10:59:21 +0300
Subject: [PATCH] Amend the RP1 ethernet node to work with upstream driver

Rebased onto raspberrypi/linux stable_20260724. The bcm2712-rpi-cm5.dtsi hunk
was refreshed: upstream added pinctrl-names/pinctrl-0 = <&eth0_irq_pins>,
renumbered the PHY from ethernet-phy@1 (reg 0x1) to ethernet-phy@0 (reg 0x0),
and changed interrupt-parent from &gpio to &rp1_gpio. The bcm2712-rpi-5-b.dts
hunk still matches upstream and is unchanged.
---
 .../boot/dts/broadcom/bcm2712-rpi-5-b.dts      | 14 ++++++++++----
 .../boot/dts/broadcom/bcm2712-rpi-cm5.dtsi     | 20 +++++++++++++-------
 2 files changed, 23 insertions(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
index 2939214..a2c3450 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
@@ -169,12 +169,18 @@ rp1_target: &pcie2 {
 &rp1_eth {
 	status = "okay";
 	phy-handle = <&phy1>;
-	phy-reset-gpios = <&rp1_gpio 32 GPIO_ACTIVE_LOW>;
-	phy-reset-duration = <5>;

-	phy1: ethernet-phy@1 {
+	mdio {
 		reg = <0x1>;
-		brcm,powerdown-enable;
+		reset-gpios = <&rp1_gpio 32 GPIO_ACTIVE_LOW>;
+		reset-delay-us = <5000>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		phy1: ethernet-phy@1 {
+			reg = <0x1>;
+			brcm,powerdown-enable;
+		};
 	};
 };

diff --git a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
index 4eb7818..18e8157 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
@@ -156,16 +156,22 @@ rp1_target: &pcie2 {
 &rp1_eth {
 	status = "okay";
 	phy-handle = <&phy1>;
-	phy-reset-gpios = <&rp1_gpio 32 GPIO_ACTIVE_LOW>;
-	phy-reset-duration = <5>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&eth0_irq_pins>;

-	phy1: ethernet-phy@0 {
+	mdio {
 		reg = <0x0>;
-		brcm,powerdown-enable;
-		interrupt-parent = <&rp1_gpio>;
-		interrupts = <37 IRQ_TYPE_LEVEL_LOW>;
+		reset-gpios = <&rp1_gpio 32 GPIO_ACTIVE_LOW>;
+		reset-delay-us = <5000>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		phy1: ethernet-phy@0 {
+			reg = <0x0>;
+			brcm,powerdown-enable;
+			interrupt-parent = <&rp1_gpio>;
+			interrupts = <37 IRQ_TYPE_LEVEL_LOW>;
+		};
 	};
 };

--
2.35.3
